securesystemslib/interface.py

```diff
@@ -120,107 +104,141 @@ def get_password(prompt='Password: ', confirm=False):
+  may change over time."  The private key is written in PKCS#1 and the public
+  key in X.509 SubjectPublicKeyInfo format.
+
+  NOTE: A signing scheme can be assigned on key import (see import functions).
+
+  Arguments:
+    filepath (optional): The path to write the private key to. If not passed,
+        the key is written to CWD using the keyid as filename. The public key
+        is written to the same path as the private key using the suffix '.pub'.
+    bits (optional): The number of bits of the generated RSA key.
+    password (optional): An encryption password.
+    prompt (optional): A boolean indicating if the user should be prompted
+        for an encryption password. If the user enters an empty password, the
+        key is not encrypted.
 
-  # Does 'bits' have the correct format?
-  # Raise 'securesystemslib.exceptions.FormatError' if there is a mismatch.
+  Raises:
+    UnsupportedLibraryError: pyca/cryptography is not available.
+    FormatError: Arguments are malformed.
+    ValueError: An empty string is passed as 'password', or both a 'password'
+        is passed and 'prompt' is true.
+    StorageError: Key files cannot be written.
+
+  Side Effects:
+    Prompts user for a password if 'prompt' is True.
+    Writes key files to disk.
+
+  Returns:
+    The private key filepath.
+
+  """
   securesystemslib.formats.RSAKEYBITS_SCHEMA.check_match(bits)
 
-  # Generate the public and private RSA keys.
+  # Generate private RSA key and extract public and private both in PEM
   rsa_key = securesystemslib.keys.generate_rsa_key(bits)
   public = rsa_key['keyval']['public']
   private = rsa_key['keyval']['private']
 
+  # Use passed 'filepath' or keyid as file name
   if not filepath:
     filepath = os.path.join(os.getcwd(), rsa_key['keyid'])
 
-  else:
-    logger.debug('The filepath has been specified.  Not using the key\'s'
-        ' KEYID as the default filepath.')
-
-  # Does 'filepath' have the correct format?
   securesystemslib.formats.PATH_SCHEMA.check_match(filepath)
 
-  # If the caller does not provide a password argument, prompt for one.
-  if password is None:
-
-    # It is safe to specify the full path of 'filepath' in the prompt and not
-    # worry about leaking sensitive information about the key's location.
-    # However, care should be taken when including the full path in exceptions
-    # and log files.
-    password = get_password('Enter a password for the encrypted RSA'
-        ' key (' + TERM_RED + filepath + TERM_RESET + '): ',
-        confirm=True)
+  password = _get_key_file_encryption_password(password, prompt, filepath)
 
-  else:
-    logger.debug('The password has been specified.  Not prompting for one')
-
-  # Does 'password' have the correct format?
-  securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
-
-  # Encrypt the private key if 'password' is set.
-  if len(password):
+  # Encrypt the private key if a 'password' was passed or entered on the prompt
+  if password is not None:
```

Comment:
Maybe we can do the password schema check here instead of forcing it on the direct/indirect caller?

Reply:
We check in the `_get_key_file_encryption_password` helper, which we call right before.